Controller/DemoController.php

```diff
@@ -21,6 +21,134 @@
+     *
+     * @param int $locationId of a folder
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function showSubFolderLinkAction( $locationId )
+    {
+        $response = new Response();
+
+        $response->setSharedMaxAge( $this->getConfigResolver()->getParameter( 'content.default_ttl' ) );
+
+        $repository = $this->getRepository();
+        $location = $repository->getLocationService()->loadLocation( $locationId );
+        if ( $location->invisible )
+        {
+            throw new NotFoundHttpException( "Location #$locationId cannot be displayed as it is flagged as invisible." );
+        }
+
+        $languages = $this->getConfigResolver()->getParameter( 'languages' );
+
+        $subContentCriteria = $this->get( 'ezdemo.criteria_helper' )->generateSubContentCriterion(
+            $location, $languages
+        );
+
+        $subContentQuery = new Query();
+        $subContentQuery->criterion = $subContentCriteria;
+        $subContentQuery->sortClauses = array(
+            new SortClause\ContentName()
+        );
+
+        $searchService = $this->getRepository()->getSearchService();
+        $subContent = $searchService->findContent( $subContentQuery );
+
+        foreach ( $subContent->searchHits as $hit )
+        {
+            $subFolders[] = $hit->valueObject;
+        }
+
+        return $this->render(
+            'eZDemoBundle:aside:folder.html.twig',
+            array(
+                'location' => $location,
+                'subFolders' => $subFolders
+            ),
+            $response
+        );
+    }
+
+    /**
+     * Displays the list of article
+     *
+     * @param int $locationId of a folder
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function showFolderListAction( $locationId )
+    {
+        $response = new Response();
+
+        $response->setSharedMaxAge( $this->getConfigResolver()->getParameter( 'content.default_ttl' ) );
+
+        $repository = $this->getRepository();
+        $location = $repository->getLocationService()->loadLocation( $locationId );
+        if ( $location->invisible )
+        {
+            throw new NotFoundHttpException( "Location #$locationId cannot be displayed as it is flagged as invisible." );
+        }
+
+        $content = $repository
+            ->getContentService()
+            ->loadContentByContentInfo( $location->getContentInfo() );
+
+        // Getting language for the current siteaccess
+        $languages = $this->getConfigResolver()->getParameter( 'languages' );
+
+        // Using the criteria helper (a demobundle custom service) to generate our query's criteria.
+        // This is a good practice in order to have less code in your controller.
+        $criteria = $this->get( 'ezdemo.criteria_helper' )->generateListFolderCriterion(
+            $location, $this->container->getParameter( 'ezdemo.folder.folder_content.excluded_content_types' ), $languages
```

Comment:
edit, sorry, i was worry about the last comma, but that was because i haven't seen the end of the line.